### DISCUSSION TITLE
Are we supposed to use `lazy: true` with relations?

### DISCUSSION THREAD

**Discussion by hayksaryan** [mikro-orm/mikro-orm#4375] [Q&A] [answered]
Is the "lazy" option only for scalar properties? Is the following example wrong?

```
@OneToMany(() => Book, (author) => book.author, {
    nullable: true,
    lazy: true,
})
books = new Collection<Book>(this);
```

**Comment by B4nan** [comment, marked as answer]
Relations are lazy by default, and for 1:m it does not have any effect at all, as that does not represent any actual database column. `lazy: true` is about skipping the column from select clause, you can use it with owning side relations like m:1 or 1:1, and the result will be the FK not being selected. But again, relations are lazy by default, even if the column gets selected, you still get just a reference in place of that FK - the target entity wont be initialized.

**Reply by hayksaryan** [reply]
Thanks a lot, that clarifies it. To summarize, we almost never need to specify that flag when defining relations.

**Reply by B4nan** [reply]
Yeah it is there for scalar properties that are holding large values, like an article text, or for things you almost never want to select, like a password hash, for relations you don't really need it.
